spec.html

```diff
@@ -37846,17 +37882,21 @@ <h1>Atomics.compareExchange ( _typedArray_, _index_, _expectedValue_, _replaceme
         1. Let _elementSize_ be the Element Size value specified in <emu-xref href="#table-the-typedarray-constructors"></emu-xref> for _arrayTypeName_.
         1. Let _offset_ be _typedArray_.[[ByteOffset]].
         1. Let _indexedPosition_ be (_i_ &times; _elementSize_) + _offset_.
-        1. Let `compareExchange` denote a semantic function of two List of byte values arguments that returns the second argument if the first argument is element-wise equal to _expectedBytes_.
-        1. Return GetModifySetValueInBuffer(_buffer_, _indexedPosition_, _elementType_, _replacement_, `compareExchange`).
+        1. Let _compareExchange_ be a new read-modify-write modification function with parameters (_oldBytes_, _newBytes_) that captures _expectedBytes_ and performs the following steps atomically when called:
+          1. If _oldBytes_ is element-wise equal to _expectedBytes_, then
+            1. Return _newBytes_.
+          1. Return _oldBytes_.
+        1. Return GetModifySetValueInBuffer(_buffer_, _indexedPosition_, _elementType_, _replacement_, _compareExchange_).
       </emu-alg>
     </emu-clause>
 
     <emu-clause id="sec-atomics.exchange">
       <h1>Atomics.exchange ( _typedArray_, _index_, _value_ )</h1>
-      <p>Let `second` denote a semantic function of two List of byte values arguments that returns its second argument.</p>
       <p>The following steps are taken:</p>
       <emu-alg>
-        1. Return ? AtomicReadModifyWrite(_typedArray_, _index_, _value_, `second`).
+        1. Let _compareExchange_ be a new read-modify-write modification function with parameters (_oldBytes_, _newBytes_) that captures nothing and performs the following steps atomically when called:
```

Comment:
can "that captures nothing" be the default state, so we don't have to state it explicitly? (or do we prefer stating it explicitly)

Reply:
Both @bakkot and I prefer stating it explicitly.